.github/workflows/e2e.yml

```diff
@@ -0,0 +1,207 @@
+name: e2e
+
+on:
+  push:
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+env:
+  GO_VERSION: "1.17.3"
+  K8S_VERSION: "v1.22.2"
+  KIND_CLUSTER_NAME: "kpng"
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GO_VERSION }}
+      id: go
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: |
+        docker build -t kpng:test -f Dockerfile .
+        mkdir _output
+        docker save kpng:test > _output/kpng-image.tar
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: test-image
+        path: _output/kpng-image.tar
+
+  e2e:
+    name: e2e
+    runs-on: ubuntu-20.04
+    timeout-minutes: 100
+    needs:
+      - build
+    strategy:
+      fail-fast: false
+      matrix:
+        ipFamily: ["ipv4", "ipv6", "dual"]
+        backend: ["iptables", "nft"]
+    env:
+      JOB_NAME: "kpng-e2e-${{ matrix.ipFamily }}-${{ matrix.backend }}"
+      IP_FAMILY: ${{ matrix.ipFamily }}
+      BACKEND: ${{ matrix.backend }}
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Enable ipv4 and ipv6 forwarding
+      run: |
+        sudo sysctl -w net.ipv6.conf.all.forwarding=1
+        sudo sysctl -w net.ipv4.ip_forward=1
```

Comment:
@jayunit100 how we handle this with our kind cluster for local tests? 
I don't want to mess around with people's docker settings on their actual host machines: 
https://docs.docker.com/config/daemon/ipv6/ 
Also, we would need to activate with ipv6 on the actual host machines.
https://kind.sigs.k8s.io/docs/user/configuration/#networking

My idea: we just check if the host + docker are supporting ipv6 with something like this 
```
cat /sys/module/ipv6/parameters/disable
```
and this for ipv4 ipforward
```
sysctl net.ipv4.ip_forward
```
Then we give a warning out and pass the affected tests. In that case, we should address it in the README.

Reply:
I think you can omit this if you want to distribute this, just document it or add a sanity check for failing, this is specific of this job

Reply:
agree, thanks for noticing this friedrich - our style of dev is heavily based on the local dev stuff so lets add it in a way that we know it will still work locally - and then automate running it in GH Actions

Great catch